### ISSUE TITLE
Add scaling factor support

### ISSUE THREAD

**Issue by walking-octopus** [lapce/lapce#785]
On some Linux devices, DPI awareness can break. The default scaling could be too small or too large for a screen. Electron allows for a workaround by enabling zoom with `Ctrl + +` and `Ctrl + -`. Maybe there's some way to specify a scaling factor.

**Comment by s-m-e** [comment]
[Ctrl]+[ScrollWheel] is also another classic, yes. I was just searching for this. In Atom, the scaling is separated from the UI, i.e. the code (and line numbers) get scaled up and down while the UI remains at a fixed size. I very much like this. Depending on how tired you are, the lighting conditions relative to your screen or the quality of code (or the lack thereof) I can quickly adjust the size while working (while the UI stays out of the way). No need to go to a config and set a font size.

**Comment by martinstark** [comment]
## Is your feature request related to a problem? Please describe.

I have a setup with a 4k monitor where I use individual scaling for each app on my system to make them usable (linux/wayland/xwayland). Lapce is the only app I have installed that doesn't support UI scaling through in-app configuration.

## Describe the solution you'd like

A config in settings to scale the entire UI evenly, by a factor. E.g. `UI Scale: 2.2` for 220% UI scale increase.

## Describe alternatives you've considered

Global monitor scaling is not an option due to scaling issues with xwayland. 

I can increase the font size of the editor and the UI, but this makes the text spill outside of UI elements, and padding does not scale together with the font size. Example of how the UI looks on my system:

![image](https://user-images.githubusercontent.com/901824/226686726-314cfa4b-9739-44fc-b4f4-4de25ea9b20c.png)

Compared to how pretty it looks with proper padding in the sample screenshots on your website:

![image](https://user-images.githubusercontent.com/901824/226686189-6cc21b62-1a5e-42da-b3a2-e0535b2e30d1.png)

**Comment by hbina** [comment]
I am not sure if it is worth it to solve this issue in druid since lapce is moving to floem when its ready. I am not sure how floem will handle this but its one of the things that needs to be considered I guess.

**Comment by dzhou121** [comment]
zoom in/out feature is added in nightly version.

**Comment by burrbull** [comment]
> zoom in/out feature is added in nightly version.

Thanks. `Ctrl + -` `Ctrl + =` works for me now.

Although settings reset after lapce restart.
Also a little glitch. After zoom changing `Settings button` popup menu appears in wrong position.

**Comment by db47h** [comment]
TL;DR: Please force the rendering toolkit to use 96 dpi (or higher if high dpi can "safely" be detected), and allow users to scale the UI.

This is caused by the rendering toolkit pulling the screen DPI from the monitor's EDID (via RandR on Linux) and a monitor that reports an incorrect physical screen size. This happens to a lot of people and correcting the issue on the end user's side requires a lot of fiddling with monitor EDIDs, kernel params and/or Xorg config. Even for a power user, it's a real PITA to get it right (and correctly identify the problem in the first place!). That's probably why Gnome, KDE and the like all default to 96 dpi (or what the X server says, which is not necessarily what RandR says...), then offer various knobs to tweak font/UI size.
